imap_processing/hi/l1c/hi_l1c.py

```diff
@@ -0,0 +1,221 @@
+"""IMAP-HI l1c processing module."""
+
+import logging
+from pathlib import Path
+from typing import Union
+
+import numpy as np
+import xarray as xr
+
+from imap_processing import imap_module_directory
+from imap_processing.cdf.cdf_attribute_manager import CdfAttributeManager
+from imap_processing.cdf.utils import load_cdf
+
+logger = logging.getLogger(__name__)
+
+
+def hi_l1c(dependencies: list, data_version: str):
+    """
+    High level IMAP-Hi l1c processing function.
+
+    This function will be expanded once the l1c processing is better defined. It
+    will need to add inputs such as Ephemerides, Goodtimes inputs, and
+    instrument status summary and will output a Pointing Set CDF as well as a
+    Goodtimes list (CDF?).
+
+    Parameters
+    ----------
+    dependencies : list
+        Input dependencies needed for l1c processing.
+
+    data_version : str
+        Data version to write to CDF files and the Data_version CDF attribute.
+        Should be in the format Vxxx
+
+    Returns
+    -------
+    processed_data : xarray.Dataset
+        Processed xarray dataset
+    """
+    logger.info("Running Hi l1c processing")
+
+    # TODO: I am not sure what the input for Goodtimes will be so for now,
+    #    If the input is a CDF, do pset processing
+    if len(dependencies) == 1 and str(dependencies[0]).endswith("cdf"):
+        l1c_dataset = generate_pset_dataset(dependencies[0])
+    else:
+        raise NotImplementedError(
+            "Input dependencies not recognized for l1c pset processing."
+        )
+
+    # TODO: revisit this
+    l1c_dataset.attrs["Data_version"] = data_version
+    return l1c_dataset
+
+
+def generate_pset_dataset(l1b_de_path: Union[Path, str]) -> xr.Dataset:
+    """
+    Process IMAP-Hi l1b product into a l1c pset xarray dataset.
+
+    Parameters
+    ----------
+    l1b_de_path : Union[Path, str]
+        Location of IMAP-Hi l1b de product.
+
+    Returns
+    -------
+    xarray.Dataset ready to be written to CDF.
+    """
+    de_dataset = load_cdf(l1b_de_path)
+    sensor_str = de_dataset.attrs["Logical_source"].split("_")[-1].split("-")[0]
+    n_esa_step = de_dataset.esa_step.data.size
+    pset_dataset = allocate_pset_dataset(n_esa_step, sensor_str)
+    # TODO: Stored epoch value needs to be consistent across ENA instruments.
+    #    SPDF says this should be the center of the time bin, but instrument
+    #    teams may disagree.
```

Comment:
For some reason I thought you could specify the left/center/right accumulation bin, but today I learned!
> For ISTP the time value of a record refers to the center of the accumulation period for the record if the measurement is not an instantaneous one.